### PR TITLE
xournalpp: 1.1.3 -> 1.2.0

### PR DESCRIPTION
**Pull request by cyclic-pentane**
## Description of changes
Added alsaLib because of missing libasound.so

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/xournalpp/default.nix [hunk 1]
```diff
@@ -23,18 +24,20 @@
...
     rev = "v${version}";
-    sha256 = "sha256-Hn7IDnbrmK3V+iz8UqdmHRV2TS4MwYSgYtnH6igbGJ8=";
+    sha256 = "sha256-0xsNfnKdGl34qeN0KZbII9w6PzC1HvvO7mtlNlRvUqQ=";
   };
 
   nativeBuildInputs = [ cmake gettext pkg-config wrapGAppsHook ];
   buildInputs =
-    [ glib
+    [
+      alsaLib
```
Comment (Tungsten842) [inline pkgs/applications/graphics/xournalpp/default.nix:39]:
```suggestion
      alsa-lib
```

File: pkgs/applications/graphics/xournalpp/default.nix [hunk 2]
```diff
@@ -23,18 +24,20 @@
...
+      glib
```
Comment (Tungsten842) [inline pkgs/applications/graphics/xournalpp/default.nix:40]: https://github.com/xournalpp/xournalpp/blob/release-1.2/CHANGELOG.md
Gtksourceview4 should be added as well, for improved latex editing.

**Review by Tungsten842 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/xournalpp/default.nix (hunk 2 above)
Comment (alyaeanyx) [reply pkgs/applications/graphics/xournalpp/default.nix:41]: Done, and tested that it works

File: pkgs/applications/graphics/xournalpp/default.nix (hunk 1 above)
Comment (Tungsten842) [reply pkgs/applications/graphics/xournalpp/default.nix:40]: Ofborg complains if you don't change this

Comment (alyaeanyx) [reply pkgs/applications/graphics/xournalpp/default.nix]: Oops, got lost in the second-last force-push. Fixed now

**Comment by Tungsten842** [comment]
Result of `nixpkgs-review pr 246281` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xournalpp</li>
  </ul>
</details>

**Review by Tungsten842 — APPROVED** [review]
Tested and working as expected

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 246281` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xournalpp</li>
  </ul>
</details>

**Review by sikmir — APPROVED** [review]

**Pull request merged**